plugin/plugins.go

```diff
@@ -0,0 +1,155 @@
+package plugin
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/alibaba/sealer/common"
+	"github.com/alibaba/sealer/logger"
+	v1 "github.com/alibaba/sealer/types/api/v1"
+	"github.com/alibaba/sealer/utils"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+/*
+config in PluginConfig:
+
+apiVersion: sealer.aliyun.com/v1alpha1
+kind: Plugin
+metadata:
+  name: SHELL
+spec:
+  action: PostInstall
+  on: role=master
+  data: |
+    kubectl taint nodes node-role.kubernetes.io/master=:NoSchedule
+
+Dump will dump the config to etc/redis-config.yaml file
+*/
+
+type Plugins interface {
+	Dump(clusterfile string) error
+	Run(cluster *v1.Cluster, phase Phase) error
+}
+
+type DumperPlugin struct {
+	configs     []v1.Plugin
+	clusterName string
+}
+
+func (c *DumperPlugin) GetPhasePlugin(phase Phase) []v1.Plugin {
+	configs := make([]v1.Plugin, 0)
+	for _, config := range c.configs {
+		on := Phase(config.Spec.On[5:])
+		if on == phase {
+			configs = append(configs, config)
+		}
+	}
+	return configs
+}
+
+func NewPlugins(clusterName string) *DumperPlugin {
```

Comment:
NewPlugins should return Plugins interface

Reply:
I tried to return the interface, but I got an error message that could not be converted